Source/SwiftLintFramework/Rules/AttributesRule.swift

```diff
@@ -0,0 +1,224 @@
+            violations = []
+        }
+
+        return validateFile(file, dictionary: dictionary) + violations
+    }
+
+    private func validateTestableImport(file: File) -> [StyleViolation] {
+        let pattern = "@testable[\n]+\\s*import*"
+        let excludingKinds = SyntaxKind.commentAndStringKinds()
+        return file.matchPattern(pattern, excludingSyntaxKinds: excludingKinds).map {
+            StyleViolation(ruleDescription: self.dynamicType.description,
+                severity: self.configuration.severity,
+                location: Location(file: file, byteOffset: $0.location))
+        }
+    }
+
+    private func validateKind(file: File,
+                              attributeShouldBeOnSameLine: Bool,
+                              dictionary: [String: SourceKitRepresentable]) -> [StyleViolation] {
+        let attributes = parseAttributes(dictionary)
+
+        guard !attributes.isEmpty,
+            let offset = (dictionary["key.offset"] as? Int64).flatMap({ Int($0) }),
+            let (lineNumber, _) = file.contents.lineAndCharacterForByteOffset(offset) else {
+            return []
+        }
+
+        let line = file.lines[lineNumber - 1]
+
+        let tokens = file.syntaxMap.tokensIn(line.byteRange)
+        let attributesTokens = tokens.flatMap { attributeName($0, file: file) }
+        let isViolation: Bool
+
+        do {
+            let previousAttributes = try attributesFromPreviousLines(lineNumber - 1, file: file)
+
+            let alwaysInSameLineAttributes = ["@IBAction", "@NSManaged"]
+            let alwaysInNewLineAttributes = ["@available"]
+
+            if !previousAttributes.filter(alwaysInNewLineAttributes.contains).isEmpty {
+                isViolation = false
+            } else if !attributesTokens.filter(alwaysInNewLineAttributes.contains).isEmpty {
+                isViolation = true
+            } else if !previousAttributes.filter(alwaysInSameLineAttributes.contains).isEmpty {
+                isViolation = true
+            } else if !attributesTokens.filter(alwaysInSameLineAttributes.contains).isEmpty {
+                isViolation = false
+            } else {
+                isViolation = attributesTokens.isEmpty == attributeShouldBeOnSameLine
+            }
+        } catch {
+            isViolation = true
+        }
+
+        guard isViolation else {
+            return []
+        }
+
+        // Violation found!
+        let location: Location
+        if let offset = (dictionary["key.offset"] as? Int64).flatMap({ Int($0) }) {
+            location = Location(file: file, byteOffset: offset)
+        } else {
+            location = Location(file: file.path)
+        }
+
+        return [
+            StyleViolation(ruleDescription: self.dynamicType.description,
+                severity: configuration.severity,
+                location: location
+            )
+        ]
+    }
+
+    private func attributesFromPreviousLines(lineNumber: Int, file: File) throws -> [String] {
```

Comment:
I don't particularly like this, but it was the only way I figured out to get the texts for the attributes, as there aren't exposed on AST: https://github.com/realm/SwiftLint/issues/846#issuecomment-254016737